Loop/View Models/BolusEntryViewModel.swift

```diff
@@ -198,7 +201,26 @@ final class BolusEntryViewModel: ObservableObject {
                 presentAlert(.manualGlucoseEntryOutOfAcceptableRange)
                 return
             }
+        }
 
+        // Authenticate the bolus before saving anything
+        if enteredBolus.doubleValue(for: .internationalUnit()) > 0 {
+            let message = String(format: NSLocalizedString("Authenticate to Bolus %@ Units", comment: "The message displayed during a device authentication prompt for bolus specification"), enteredBolusAmountString)
+            authenticate(message) {
+                switch $0 {
+                case .success:
+                    self.continueSaving(onSuccess: completion)
+                case .failure:
+                    break
+                }
+            }
+        } else if potentialCarbEntry != nil  { // Allow user to save carbs without bolusing
+            self.continueSaving(onSuccess: completion)
+        }
```

Comment:
else...`completion()`?
```suggestion
        } else {
            completion()
        }
```

Reply:
(I committed this change directly, and this is still showing up, but it's in there)